nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -425,67 +444,68 @@ class MinReducer(TensorReducerBase):
         reduction_shape = self._get_reduction_shape(x)
-        return [self._tensor_processor.reduce_min(x, reduction_shape, keepdims=True)]
+        return [self._tensor_processor.reduce_min(x, reduction_shape, keepdims=self._keepdims)]
 
 
 class MaxReducer(TensorReducerBase):
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
         x = x[0]
         reduction_shape = self._get_reduction_shape(x)
-        return [self._tensor_processor.reduce_max(x, reduction_shape, keepdims=True)]
+        return [self._tensor_processor.reduce_max(x, reduction_shape, keepdims=self._keepdims)]
 
 
 class AbsMaxReducer(TensorReducerBase):
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
         x = self._tensor_processor.abs(x[0])
         reduction_shape = self._get_reduction_shape(x)
-        return [self._tensor_processor.reduce_max(x, reduction_shape, keepdims=True)]
+        return [self._tensor_processor.reduce_max(x, reduction_shape, keepdims=self._keepdims)]
 
 
 class MeanReducer(TensorReducerBase):
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
         x = x[0]
         reduction_shape = self._get_reduction_shape(x)
-        return [self._tensor_processor.mean(x, reduction_shape, keepdims=True)]
+        return [self._tensor_processor.mean(x, reduction_shape, keepdims=self._keepdims)]
 
 
 class QuantileReducerBase(TensorReducerBase):
     def __init__(
         self,
-        reduction_shape: Optional[ReductionShape] = None,
+        reduction_axes: Optional[ReductionShape] = None,
         quantile: Optional[Union[float, Tuple[float]]] = None,
         inplace: bool = False,
     ):
-        super().__init__(reduction_shape, False)
+        super().__init__(reduction_axes=reduction_axes, inplace=False)
         self._quantile = (0.01, 0.99) if quantile is None else quantile
 
     def __eq__(self, __o: object) -> bool:
         return super().__eq__(__o) and self._quantile == __o._quantile
 
     def __hash__(self) -> int:
-        return hash((self.__class__.__name__, self.inplace, self._reduction_shape, tuple(self._quantile)))
+        return hash((self.__class__.__name__, self.inplace, self._reduction_axes, tuple(self._quantile)))
 
 
 class QuantileReducer(QuantileReducerBase):
     def _reduce_out_of_place(self, x: List[NNCFTensor]) -> List[NNCFTensor]:
         x = x[0]
         reduction_shape = self._get_reduction_shape(x)
-        return self._tensor_processor.quantile(x, self._quantile, reduction_shape, keepdims=True)
+        return self._tensor_processor.quantile(x, self._quantile, reduction_shape, keepdims=self._keepdims)
 
 
 class AbsQuantileReducer(QuantileReducerBase):
     def __init__(
         self,
-        reduction_shape: Optional[ReductionShape] = None,
-        quantile: Union[float, List[float]] = 0.99,
+        reduction_axes: Optional[ReductionShape] = None,
+        quantile: Optional[Union[float, List[float]]] = None,
```

Comment:
Because it has default values. Do you think we don't need this?

Reply:
0.99 doesn't seem to be a value which could be considered "default", and if one requests quantiles, then they are most likely requesting a specific quantile, not just any good-looking quantile.

Reply:
This default value is used in ChannelAlignment, 0.99 is default quantile to find outliers in data

Reply:
Then the `ChannelAlignment` algo will instantiate the collector with exact specification of 0.99 in its own code, so that it is immediately obvious that 0.99 is a magic value designated by `ChannelAlignment` specifically, not by the collector.

Reply:
Valid point. But as it was optional before (in `QuantileReducerBase`  and `NoOutlierReducerBase`), I suggest to put this request to a refactoring request list, what do you think?

Reply:
Ok